src/NuGet.Clients/PackageManagement.VisualStudio/SolutionRestore/SolutionRestoreWorker.cs

```diff
@@ -91,21 +92,25 @@ internal sealed class SolutionRestoreWorker : SolutionEventsListener, ISolutionR
             Reset();
         }
 
-        public async Task InitializeAsync(IAsyncServiceProvider site)
+        private async Task InitializeAsync()
         {
-            await _joinableFactory.RunAsync(async () =>
+            if (!_initialized)
```

Comment:
I thought about that but then saw it will never be used in multi-thread scenario... But anyways I'll make it thread-safe to be on safer side.